apps/testingaccessibility/src/pages/thanks/purchase.tsx

```diff
@@ -25,14 +32,54 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
     }
   }
 
-  const checkoutSession = await getCheckoutSession(session_id as string)
+  const purchaseInfo = await stripeData(session_id as string)
 
-  const {customer, line_items} = checkoutSession
-  const {email} = customer as Stripe.Customer
+  const {email, stripeChargeId, quantity: seatsPurchased} = purchaseInfo
 
-  const {quantity: seatsPurchased} = line_items?.data[0] || {quantity: 1}
+  // Find the purchase associated with this stripeChargeId
+  // - Is the tied to a bulk coupon?
+  //   - If so, is it the first one?
+  //     - Yes: NEW_BULK_COUPON
+  //     - No:  EXISTING_BULK_COUPON
+  //   - If not: NEW_INDIVIDUAL_PURCHASE
 
-  const validatedProps = thanksProps.parse({email, seatsPurchased})
+  const purchase = await prisma.purchase.findFirst({
+    where: {
+      merchantCharge: {
+        identifier: stripeChargeId,
+      },
+    },
+    include: {
+      bulkCoupon: {
+        include: {
+          bulkCouponPurchases: true,
+        },
+      },
+    },
+  })
+
+  if (!purchase || !email) {
+    return {
+      notFound: true,
+    }
+  }
+
+  let purchaseType: z.infer<typeof purchaseTypeSchema>
+  if (purchase.bulkCoupon) {
+    if (purchase.bulkCoupon.bulkCouponPurchases.length > 1) {
+      purchaseType = EXISTING_BULK_COUPON
+    } else {
+      purchaseType = NEW_BULK_COUPON
+    }
+  } else {
+    purchaseType = NEW_INDIVIDUAL_PURCHASE
+  }
```

Comment:
Determine what type of purchase this checkout session was for. This `purchaseType` value can be used by the component to adjust the copy accordingly.